docker/Dockerfile

```diff
@@ -33,22 +29,25 @@ RUN wget https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.5.tar.g
 ENV PATH=$PATH:/usr/local/openmpi/bin
 ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/openmpi/lib
 
+RUN --mount=type=cache,target=/root/.cache/pip python3 -m pip install --upgrade pip setuptools==69.5.1 &&\
+    python3 -m pip install torch==${TORCH_VERSION} torchvision==${TORCHVISION_VERSION} --index-url https://download.pytorch.org/whl/${CUDA_VERSION_SHORT} &&\
+    python3 -m pip install cmake packaging wheel
+
 ENV NCCL_LAUNCH_MODE=GROUP
 
 # Should be in the lmdeploy root directory when building docker image
 COPY . /opt/lmdeploy
 
 WORKDIR /opt/lmdeploy
 
-RUN cd /opt/lmdeploy &&\
-    python3 -m pip install --no-cache-dir -r requirements.txt &&\
+RUN --mount=type=cache,target=/root/.cache/pip cd /opt/lmdeploy &&\
+    python3 -m pip install -r requirements.txt &&\
     mkdir -p build && cd build &&\
     sh ../generate.sh &&\
     ninja -j$(nproc) && ninja install &&\
     cd .. &&\
     python3 -m pip install -e . &&\
-    rm -rf build  &&\
-    rm -rf ~/.cache/*
```

Comment:
Shoudl we keep `rm -rf ~/.cache/*?
One contributor said it helps reduce the image size

Reply:
No need. While building docker image, pip wheels are caching onto the host machine. You can verify by 
```
docker run -it --rm openmmlab/lmdeploy:latest-cu12 python3 -m pip cache info
```
 outputs
```
Package index page cache location (pip v23.3+): /root/.cache/pip/http-v2
Package index page cache location (older pips): /root/.cache/pip/http
Package index page cache size: 0 bytes
Number of HTTP files: 0
Locally built wheels location: /root/.cache/pip/wheels
Locally built wheels size: 0 bytes
Number of locally built wheels: 0
```

Reply:
Will the cache affect the upgrade of dependencies?
For example, a new release of a dependency repo is public, will it be used during lmdeploy docker image build?

Reply:
No effect if the package is upgraded due to a new release. Besides, the cache may not work for building docker image in CI since the github hosted machines may delete the cache automatically.  But it could definitely speed up building image on our  local machine.